core/src/main/java/org/elasticsearch/http/HttpServer.java

```diff
@@ -117,4 +124,24 @@ void handleFavicon(HttpRequest request, HttpChannel channel) {
             channel.sendResponse(new BytesRestResponse(FORBIDDEN));
         }
     }
+
+    private static final class ResourceHandlingHttpChannel extends HttpChannel {
+        private final HttpChannel delegate;
+        private final HttpServerTransport transport;
+
+        public ResourceHandlingHttpChannel(HttpChannel delegate, HttpServerTransport transport) {
+            super(delegate.request(), delegate.detailedErrorsEnabled());
```

Comment:
this is not good...  you rely on the base class implementation (that only these two parameters for it's state). You have to proxy all requests to the delegate